pkgs/development/haskell-modules/default.nix

```diff
@@ -7,26 +7,24 @@
 , configurationNix ? import ./configuration-nix.nix
 }:
 
+self: # Provided by `callPackageWithSelf`
```

Comment:
Yea. Since `self` becomes the scope for all the packages' `overrideScope`, it needs to support `extend`, thus it needs to be the final output, not the internal one.